[H-ZhanHao] iP

H-ZhanHao opened the pull request.

Zhoop321 committed:
Change Duke to Eelyon

H-ZhanHao committed:
Update Eelyon to add echo feature

H-ZhanHao committed:
Update Eelyon: Add ability to store text into list

H-ZhanHao committed:
Add Task.java
Update Eelyon: Add ability to unmark and mark as done

H-ZhanHao committed:
Add Deadline.java, Event.java, Todo.java.

Eelyon updated to handle deadlines, events, and todos

H-ZhanHao committed:
Bug Fix: Can now view mark and unmarked tasks

shira421 commented on src/main/java/Eelyon.java:
```diff
@@ -0,0 +1,111 @@
+    public static void main(String[] args) {
+        boolean isFinished = false;
+        String input;
+        String[] seperatedInput;
```
separated is spelt with an a

shira421 commented on src/main/java/Eelyon.java:
```diff
@@ -0,0 +1,111 @@
+            seperatedInput = input.trim().split(" ");
+
+            switch (seperatedInput[0]) {
+            case "mark":
```
The indentation for the switch statement is correct. Nice!

shira421 commented on src/main/java/Eelyon.java:
```diff
@@ -0,0 +1,111 @@
+            System.out.println("Invalid index");
+            System.out.println(LINE);
+            return;
+        } else {
```
else statement is on the correct line. Nice!

shira421 commented on src/main/java/Eelyon.java:
```diff
@@ -0,0 +1,111 @@
+
+    public static void addTask(Task task) {
+        list[listIndex] = task;
+        System.out.println(LINE + "added: " + list[listIndex].description + "\n");
```
Could this be written as task.description instead of list[listIndex].description?

olsonwangyj commented on src/main/java/Eelyon.java:
```diff
@@ -0,0 +1,111 @@
+        if (index >= listIndex || index < 0) {
+            System.out.println(LINE);
+            System.out.println("Invalid index");
+            System.out.println(LINE);
```
maybe can extract methods for the error handle

olsonwangyj commented on src/main/java/Eelyon.java:
```diff
@@ -0,0 +1,111 @@
+        if (index >= listIndex || index < 0) {
+            System.out.println(LINE);
+            System.out.println("Invalid index");
+            System.out.println(LINE);
```
maybe can extract methods for the error handle

olsonwangyj commented on src/main/java/Eelyon.java:
```diff
@@ -0,0 +1,111 @@
+                break;
+            default:
+                System.out.println("Invalid input");
+                break;
```
at the same high level of the abstraction

olsonwangyj commented on src/main/java/Eelyon.java:
```diff
@@ -0,0 +1,111 @@
+
+
+public class Eelyon {
+    static Task[] list = new Task[100];
```
maybe can avoid using the magic number

AnotherSACE commented on src/main/java/Eelyon.java:
```diff
@@ -0,0 +1,111 @@
+                System.out.println("Invalid input");
+                break;
+            }
+        }
```
Switch case is very readable!

olsonwangyj commented on src/main/java/Eelyon.java:
```diff
@@ -0,0 +1,111 @@
+    public static void addTask(Task task) {
+        list[listIndex] = task;
+        System.out.println(LINE + "added: " + list[listIndex].description + "\n");
+        listIndex++;
```
maybe can struct the code more logically

olsonwangyj reviewed (commented):
overall very good coding quality

AnotherSACE commented on src/main/java/Deadline.java:
Looks good and easy to read!

AnotherSACE commented on src/main/java/Task.java:
Looks good, easy to read

AnotherSACE commented on src/main/java/Todo.java:
Easy to read, looks good

AnotherSACE commented on src/main/java/Event.java:
Looks good!

AnotherSACE reviewed (commented):
Everything looks very easy to read! Small issues have already been addressed previously

H-ZhanHao committed:
Add CammandType.java
Main.java: Add error handling

System would crash whenever an input was unknown or format was incorrect

Implemented a new function called inputCheck to check for any descriptions present as well as correct formatting.
CommandType class was created to abstract the different formats of commands.

Using objects made the code cleaner and reduced repetitive code.

H-ZhanHao committed:
Merge branch 'branch-Level-5'

H-ZhanHao committed:
Update Eelyon: add delete task function

H-ZhanHao committed:
Update Eelyon: add saving and loading

H-ZhanHao committed:
Merge branch 'branch-Level-6'

H-ZhanHao committed:
Merge branch 'branch-Level-7'

# Conflicts:
#	src/main/java/Eelyon/Eelyon.java

H-ZhanHao committed:
Fix bugs: Merge conflict solved

H-ZhanHao committed:
Bug fix: Directory now created if it doesn't exist

sevenseasofbri commented on src/main/java/Eelyon/Commands/CommandType.java:
```diff
@@ -0,0 +1,44 @@
+
+    public void setCommandType(String commandType) {
+        switch(commandType) {
+        case "deadline":
+        case "event":
+        case "todo":
+        case "mark":
+        case "unmark":
+        case "list":
+        case "delete":
```
> From the java coding standard: The explicit `// Fallthrough` comment should be included whenever there is a case statement without a break statement.

sevenseasofbri commented on src/main/java/Eelyon/Eelyon.java:
```diff
@@ -0,0 +1,296 @@
+package Eelyon;
+
+import Eelyon.Commands.*;
```
Avoid using wildcard imports. Import the classes used explicitly.

sevenseasofbri commented on src/main/java/Eelyon/Eelyon.java:
```diff
@@ -0,0 +1,296 @@
+            if (input.substring(input.indexOf("todo") + "todo".length()).trim().length() <= 0) {
+                throw new EmptyDescriptionException("No empty descriptions allowed dawg. Try again.");
+            }
+        } else if (input.substring(input.indexOf(commandType.getCommandType()) + commandType.getCommandType().length(), input.indexOf(commandType.getEndCommand())).trim().length() <= 0) {
```
Lines with >120 chars can hinder readability. The statement is also very complex. Consider breaking it into parts. Else wrap the text onto the next line. Although in this case I would suggest you make this expression less complex.

sevenseasofbri replied:
This applies to lines 103, 126, 132 too.

sevenseasofbri replied:
Notes on [avoiding complex expressions](https://nus-cs2113-ay2425s2.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-complicated-expressions)

sevenseasofbri commented on src/main/java/Eelyon/Eelyon.java:
```diff
@@ -0,0 +1,296 @@
+
+        try {
+
+            File f = new File(FILE_PATH);
```
the name `f` may not be intuitive, consider naming it something else .

sevenseasofbri commented on src/main/java/Eelyon/Eelyon.java:
```diff
@@ -0,0 +1,296 @@
+                System.out.println("Save File Created Successfully");
+            } else {
+                System.out.println("Save file already exists. Loading from file...");
+                try {
+                    loadFile(f);
+                } catch (FileNotFoundException e) {
+                    System.out.println(e.getMessage());
+                }
```
Deep nesting, you might want to refactor this to a separate function

sevenseasofbri commented on src/main/java/Eelyon/Eelyon.java:
```diff
@@ -0,0 +1,296 @@
+    }
+
+
+    public static void main(String[] args) {
```
This function is very long (>30 LOC). Break it into parts and refactor it.

sevenseasofbri reviewed (commented):
Good job so far! Readability for some sections pertaining to string formatting can be improved though.

H-ZhanHao committed:
Update Eelyon to be more OOP

H-ZhanHao committed:
Add findCommand

Users can now search for items in the list

H-ZhanHao committed:
Add Javadoc documentation

H-ZhanHao committed:
Merge pull request #1 from H-ZhanHao/branch-A-JavaDoc

Add Javadoc documentation

H-ZhanHao committed:
Merge branch 'master' into branch-Level-9

# Conflicts:
#	src/main/java/Eelyon/control/Ui.java

H-ZhanHao committed:
Merge pull request #2 from H-ZhanHao/branch-Level-9

Branch level 9

H-ZhanHao committed:
Update README
bug fix: Fix spacing issue for event tasks and parsing issue

README now includes all the important commands users need to operate Eelyon

H-ZhanHao committed:
Fix code format issues

H-ZhanHao committed:
Merge branch 'branch-Level-7'

# Conflicts:
#	src/main/java/Eelyon/Commands/Deadline.java
#	src/main/java/Eelyon/Commands/Event.java
#	src/main/java/Eelyon/Eelyon.java

H-ZhanHao committed:
Bug fix: Parsing issue with event tasks

H-ZhanHao committed:
Update Eelyon: fixed coding style issue
